### ISSUE TITLE
Select project based on CNAME record

### ISSUE THREAD

**Issue by jorisvdz** [directus/api-next#138]
# Feature Request
Hi,

Imagine having multiple projects within one Directus installation. Wouldn't it be great to be able to select a project based on a CNAME record?

For example:

- cms.website1.com will select "website1" as project
- cms.website2.com will select "website2" as project

## What problem does this feature solve?
Easier selecting a specific project, with a more simplified URL

## How do you think this should be implemented?
Maybe within a custom Apache template?

## Would you be willing to work on this?
Yes, but I am not quite sure where to start.

**Comment by rijkvanzanten** [comment]
We're actually working on something very similar for the next major version of the API. I believe we already have support for subdomain-project-selection, which should also make CNAME matching possible.

@WoLfulus

**Comment by WoLfulus** [comment]
We already have thought and implemented a flexible way to change this through `Identifiers`, but to have this working properly we need to rethink/decouple how routing works on Directus (for example /projects route returning the full project url and not just the name of it). We'll also need to change how `app` works too because all routes right now are hard coded and should live in specific levels.

**Comment by rijkvanzanten** [comment]
I don't think we should support multi-domain for the app, as that royally screws with CORS and same-origin restrictions in the browser. That being said, if we have multi-subdomain support, you can then attach whatever CNAME to the API and target the API through that means.

**Comment by rijkvanzanten** [comment]
With multi-domain I mean `example.com/admin` connecting to `api.rijks.website`

**Comment by WoLfulus** [comment]
We'll have to take a better look at this anyways when it's the time as this is a really common scenario that we didn't get right at first, but I'm pretty sure it's doable.

**Comment by rijkvanzanten** [comment]
Just be _very_ aware of https://digiday.com/media/what-is-chrome-samesite/

**Comment by WoLfulus** [comment]
Related to #139

**Comment by jooola** [comment]
I feel like using CNAME to ID a project is usefull, but too much hassle to manage for regular users.

It might be interesting to allow any CNAME to hit the directus API, but not necessarily filter project with it. Filtering should be made with something not attached to the network/server setup. A HTTP header might be something to consider, since adding/removing a project don't require to configure a new CNAME in in the netwwork/server layer.

**Comment by rijkvanzanten** [comment]
It's more of a power-user enhancement than _the_ standard way of doing things
